woocommerce/wc-admin
# Add Live Data to Reviews Panel

Fixes #1036.

This PR hooks up the 'reviews' activity panel to live data.

It also modifies the `/products/*` and `/products/reviews/*` endpoints, by allowing embedding (https://developer.wordpress.org/rest-api/using-the-rest-api/linking-and-embedding/) of some product fields when `?_embed=1` is present in the query string.

This allows us to get info like the product name, permalink, and image without extra requests. It also **matches the behavior** of WP-API Core endpoints. For example on a comment, you can receive some posts field. So this change makes things easier for us, and makes sense from a consistency point of view. (You could already embed some user info via the reviews endpoint).

### Screenshots

<img width="845" alt="screen shot 2018-12-10 at 1 25 49 pm" src="https://user-images.githubusercontent.com/689165/49759260-2836c100-fc8f-11e8-8072-5cc82bcf976c.png">

### Detailed test instructions:

- Run `phpunit` and make sure all tests pass
- Click reviews on the activity panel
- Verify that your reviews properly load in
